src/blas/backends/rocblas/rocblas_level3.cpp

```diff
@@ -0,0 +1,1365 @@
+                               lda, b_, ldb, (rocDataType *)&beta, c_, ldc);
+        });
+    });
+    return done;
+}
+
+#define GEMM_LAUNCHER_USM(TYPE, ROCBLAS_ROUTINE)                                                  \
+    cl::sycl::event gemm(cl::sycl::queue &queue, transpose transa, transpose transb, int64_t m,   \
+                         int64_t n, int64_t k, TYPE alpha, const TYPE *a, int64_t lda,            \
+                         const TYPE *b, int64_t ldb, TYPE beta, TYPE *c, int64_t ldc,             \
+                         const cl::sycl::vector_class<cl::sycl::event> &dependencies) {           \
+        return gemm(ROCBLAS_ROUTINE, queue, transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, \
+                    c, ldc, dependencies);                                                        \
+    }
+
+GEMM_LAUNCHER_USM(float, rocblas_sgemm)
+GEMM_LAUNCHER_USM(double, rocblas_dgemm)
+GEMM_LAUNCHER_USM(std::complex<float>, rocblas_cgemm)
+GEMM_LAUNCHER_USM(std::complex<double>, rocblas_zgemm)
+
+#undef GEMM_LAUNCHER_USM
+template <typename Func, typename T_A, typename T_B, typename T_C, typename DATATYPE_A,
+          typename DATATYPE_B, typename DATATYPE_C>
+inline cl::sycl::event gemm(Func func, DATATYPE_A DT_A, DATATYPE_B DT_B, DATATYPE_C DT_C,
+                            cl::sycl::queue &queue, transpose transa, transpose transb, int64_t m,
+                            int64_t n, int64_t k, T_C alpha, const T_A *a, int64_t lda,
+                            const T_B *b, int64_t ldb, T_C beta, T_C *c, int64_t ldc,
+                            const cl::sycl::vector_class<cl::sycl::event> &dependencies) {
+    using rocDataType_A = typename RocEquivalentType<T_A>::Type;
+    using rocDataType_B = typename RocEquivalentType<T_B>::Type;
+    using rocDataType_C = typename RocEquivalentType<T_C>::Type;
+    overflow_check(m, n, k, lda, ldb, ldc);
+    auto done = queue.submit([&](cl::sycl::handler &cgh) {
+        int64_t num_events = dependencies.size();
+        for (int64_t i = 0; i < num_events; i++) {
+            cgh.depends_on(dependencies[i]);
+        }
+        onemkl_rocblas_host_task(cgh, queue, [=](RocblasScopedContextHandler &sc) {
+            auto handle = sc.get_handle(queue);
+            auto a_ = reinterpret_cast<const rocDataType_A *>(a);
+            auto b_ = reinterpret_cast<const rocDataType_B *>(b);
+            auto c_ = reinterpret_cast<rocDataType_C *>(c);
+            rocblas_status err;
+            ROCBLAS_ERROR_FUNC(func, err, handle, get_rocblas_operation(transa),
+                               get_rocblas_operation(transb), m, n, k, (rocDataType_C *)&alpha, a_,
+                               DT_A, lda, b_, DT_B, ldb, (rocDataType_C *)&beta, c_, DT_C, ldc, c_,
+                               DT_C, ldc, DT_C, rocblas_gemm_algo_standard, 0,
+                               rocblas_gemm_flags_pack_int8x4);
```

Comment:
Why do we need this flag?

Reply:
According to the [release notes](https://github.com/ROCmSoftwarePlatform/rocBLAS/releases/tag/rocm-4.2.0) this is only needed for `int8` datatype. I removed this from the PR.